yarn-project/noir-libs/noir-aztec/src/note/note_getter.nr

```diff
@@ -48,6 +52,48 @@ fn ensure_note_exists<Note, N>(
     context.push_read_request(note_hash_for_read_request);
 }
 
+// Ensure a note's hash exists in the tree without retrieving the entire
+// notes via the oracle.
+// Modifies the note by populating it with header info.
+fn ensure_note_hash_exists<Note, N>(
+    context: &mut Context,
+    storage_slot: Field,
+    note_interface: NoteInterface<Note, N>,
+    note: &mut Note,
```

Comment:
I dont think note needs to be mutable here?

Reply:
The note ends up getting its header populated although the existing NonNativeToken contract does not actually use the updated note. But this mimics the `ensure_note_exists` function above it.